lesson17/script.js

```diff
@@ -0,0 +1,147 @@
+    }
+    image.className = className;
+    image.src = src;
+    image.alt = alt;
+    return image;
+}
+
+const createButton = ({id,className}) => {
+    const button = document.createElement("button");
+    button.id = id;
+    button.className = className;
+    return button;
+}
+
+const createErrorMessage = (error) => {
+    const errMsg  = createElement({tagName:"p",className:"error-msg",text:error})
+
+    slider.appendChild(errMsg);
+    return errMsg;
+}
+
+const showLoading = () => {
+    const loadingImg = createImage({
+        id:"js-loading",
+        className:"loading",
+        src:"img/loading-circle.gif",
+        alt:"ローディング"
+    });
+
+    slider.appendChild(loadingImg);
+}
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const fetchData = async (URL) => {
+    const response = await fetch(URL);
+    if(!response.ok){
+        console.error(`${response.status}:${response.statusText}`);
+        createErrorMessage('現在、サーバーの通信が壊れています');
+    }
+    const json = await response.json();
+    const data = json.data;
+    return data;
+}
+
+const fetchSliderData = (ms) => new Promise(resolve => setTimeout(() => resolve(fetchData("https://myjson.dit.upm.es/api/bins/2bzx")),ms));
+
+const init = async () => {
+    showLoading();
+    try{
+        const sliderData = await fetchSliderData(3000);
+        if (!sliderData) {
+            return;
+        }
+        renderSliderItems(sliderData);
+        renderPagenation(sliderData);
+        
+        renderButtons();
+        isActiveFirstSliderItem();
+        SwitchSliderItem();
+    }catch(error){
+        console.error(error);
+        createErrorMessage(error);
+    }finally{
+        removeLoading();
+    }
+}
+
+const renderSliderItems = (sliderData) => {
+    const fragment = document.createDocumentFragment();
+    for(const [index,image] of sliderData.entries()){
+        const sliderItem = createElement({tagName:"div",className:"slider__item"});
+        const slideImage = createImage({className:"slide-image", src:image.img, alt:image.alt});
+        sliderItem.style.zIndex = `${sliderData.length -+ index}`;
+        
+        fragment.appendChild(sliderItem).appendChild(slideImage);
+    }
+    slider.appendChild(fragment);
+}
+
+const renderButtons = () => {
+    const prevBtn = createButton({id:"js-prevBtn",className:"js-btn prev-btn slider-btn"});
+    const nextBtn = createButton({id:"js-nextBtn",className:"js-btn next-btn slider-btn"});
+    prevBtn.disabled = "true";
+
+    slider.appendChild(prevBtn).insertAdjacentElement("afterend",nextBtn);
+}
+
+const renderPagenation = (sliderData) => {
+    const pagenation    = createElement({tagName:"p",className:"pagenation",text:`/${sliderData.length}`});
+    const currentNumber = createElement({tagName:"span",id:"js-currentNumber",className:"current-number",text:"1"});
+
+    slider.appendChild(pagenation).insertAdjacentElement("afterbegin",currentNumber);
+    return pagenation;
+}
```

Comment:
IMO
I was given the same point by Moriken sensei, who told me that getPagenation is appropriate here, as it returns the element, not renderPagenation.

Reply:
I see, that's true.
I will rename.

Reply:
I renamed.
https://github.com/nakagawamai/js-lesson/pull/19/commits/9327f38ee4e93b2dc85a938d4467c7af2a2b1c6e